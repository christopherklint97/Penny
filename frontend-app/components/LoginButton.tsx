import React from 'react'
import { useAuth0 } from '@auth0/auth0-react'

const LoginButton = () => {
  const { loginWithRedirect } = useAuth0()

  return (
    <button
      className="bg-green-400 hover:bg-green-300 text-gray-900 px-3 py-1 rounded-md"
      onClick={() => loginWithRedirect()}
    >
      Login
    </button>
  )
}

export default LoginButton
