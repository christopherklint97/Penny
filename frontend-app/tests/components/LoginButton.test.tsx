import React from 'react'
import { render } from '../testUtils'
import LoginButton from '../../components/LoginButton'

describe('LoginButton component', () => {
  it('renders without crashing', function () {
    render(<LoginButton />)
  })

  it('matches snapshot', () => {
    const { asFragment } = render(<LoginButton />, {})
    expect(asFragment()).toMatchSnapshot()
  })
})
