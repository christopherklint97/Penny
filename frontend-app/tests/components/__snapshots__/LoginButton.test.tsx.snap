// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginButton component matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="bg-gray-100 hover:bg-gray-300 text-gray-900 px-3 py-1 rounded-md"
  >
    Login
  </button>
</DocumentFragment>
`;
